src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_utils.cpp

```diff
@@ -59,16 +59,19 @@ BrgemmConfig::BrgemmConfig(const dnnl::impl::cpu::x64::cpu_isa_t& isa,
       m_with_compensations(src_dt == ov::element::i8 && !one_of(m_isa, avx512_core_amx, avx2_vnni_2)),
       m_are_wei_constant(are_wei_constant),
       m_transposed_b(transposed_b),
-      m_are_wei_blocked(ov::intel_cpu::pass::BrgemmCPUBlocking::is_kn_blocking_supported(src_dt) && m_are_wei_constant),
+      m_are_wei_blocked(ov::intel_cpu::pass::BrgemmCPUBlocking::is_kn_blocking_supported(src_dt)),
       m_wei_k_blk(get_elems_in_vec(wei_dt)) {
     const auto is_fp32 = src_dt == ov::element::f32 && wei_dt == ov::element::f32;
 
     // FC always requires weight repacking
+    // BrgemmToBrgemmCPU insert BrgemmCopyB based on this flag.
+    // EliminateBrgemmCopyB remove BrgemmCopyB to outside and markup runtime config
+    // BrgemmExternalRepackingAdjuster set runtime executor for runtime config
     m_with_wei_repacking = !is_fp32 || transposed_b || m_are_wei_constant || m_are_wei_blocked;
 
     // TODO: Add more logic based on shapes and prc
     if (m_are_wei_blocked) {
-        m_wei_n_blk = is_superset(m_isa, avx512_core) ? 64 : 48;
+        m_wei_n_blk = is_superset(m_isa, avx512_core) ? 64 : 24;
```

Comment:
I suggest to split it into two initializations by `m_are_wei_constant` value because oneDNN MatMul always sets `n_blk = 24` on avx2 systems while onednn FC initializes `n_blk` by N dimension which is always constant for FC.

So in the future we will be able to adjust `m_wei_n_blk` by N, K dimensions for FullyConnected with constant second input shape.

Reply:
Also could you please leave only 64 as `default_n_blk` [here](https://github.com/openvinotoolkit/openvino/blob/32b02dacb8394e1be49be1f54aa0b5e7aa153861/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/brgemm_cpu_blocking.cpp#L120)?
I see that for now it's not used because `kn` blocking is enabled only on fp32 - and then `are_wei_blocked` is always `true`. Because of that probole it would be better if we remove some extra additional conditions.

What do you think?

Reply:
I agree. separate block size set depend on `m_are_wei_constant `, and removed default n blk condition in `BrgemmCPUBlocking`, as the value in config is used now. Thanks Alexandra!